frontend/src/pages/Videos.js

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useRef } from "react";
+import { useInfiniteQuery, useQuery } from "react-query";
+import { useSearchParams } from "react-router-dom";
+import VideoCard from "../components/VideoCard";
+import { VideosWrapper, Wrapper } from "../styles/Videos";
+import { fetchMovies } from "../utils/tmdb";
+import loaderSvg from "../assets/loader.svg";
+
+const Videos = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const params = Object.fromEntries(new URLSearchParams(searchParams));
+  const selectedGenre = useRef(params.genre ? params.genre : "popular");
+
+  const { data, isFetching, refetch } = useQuery(
+    `movies`,
+    () => fetchMovies(selectedGenre.current),
+    {
+      staleTime: Infinity,
+    }
+  );
```

Comment:
woah! learned implementation of `useQuery` here. nice!

Reply:
react-query is awesome😇